src/backend/distributed/executor/insert_select_executor.c

```diff
@@ -128,8 +128,20 @@ CoordinatorInsertSelectExecScan(CustomScanState *node)
 
 			if (prunedTaskList != NIL)
 			{
-				ExecuteMultipleTasks(scanState, prunedTaskList, isModificationQuery,
-									 hasReturning);
+				TupleDesc tupleDescriptor =
+					scanState->customScanState.ss.ps.ps_ResultTupleSlot->tts_tupleDescriptor;
+
+				if (scanState->tuplestorestate == NULL)
```

Comment:
Can this be not-NULL here?

Reply:
No, this is our own variable, assigned by `CoordinatorInsertSelectExecScan` itself.